### PR TITLE
test against nightly builds of `astropy`, `scipy`, and `numpy`

### PR DESCRIPTION
**Pull request by zacharyburnett**
Related to [SCSB-69](https://jira.stsci.edu/browse/SCSB-69)

This PR ensures that `astropy`, `scipy`, and `numpy` are all built with nightly wheels against `main`

**Checklist**
- [ ] ~added entry in `CHANGES.rst` under the corresponding subsection~
- [x] updated relevant tests
- [ ] ~updated relevant documentation~
- [ ] ~updated relevant milestone(s)~
- [x] added relevant label(s)

### CODE REVIEW

**Review by ddavis-stsci — APPROVED** [review]
LGTM, Note: once PR#693 is approved the test-oldestdeps-cov-xdist (ubuntu-latest) should pass.

**Comment by zacharyburnett** [comment]
for some reason, the installed version of `astropy` in the `devdeps` testing is still `5.2.2` instead of `5.3rc1`:
https://github.com/spacetelescope/romancal/actions/runs/4926111203/jobs/8801225303#step:10:249

I'll investigate

File: requirements-dev.txt
Comment (WilliamJamieson) [inline requirements-dev.txt:8]: This change does absolutely nothing in the context of the rest of the changes here. In fact, one can see in the CI: https://github.com/spacetelescope/romancal/actions/runs/4926111203/jobs/8801225303#step:10:265 that it is using `numpy 1.24.3` which is the latest release version of numpy NOT some "nightly-build".

If you want to use this to actually access the nightly builds (which in reality are more like weekly builds), you need to specify an alternate index for `pip` to search.

For example https://github.com/astropy/astropy/blob/337798f36c9aa2ae98190c5819bde758d3aa5a40/tox.ini#L27 specifies the "nightly-build" provided by anaconda.

File: requirements-dev.txt
Comment (WilliamJamieson) [inline requirements-dev.txt:10]: Same as `numpy` comment.

**Pull request merged**
